test/oban/job_test.exs

```diff
@@ -56,16 +56,37 @@ defmodule Oban.JobTest do
         |> Map.keys()
       end
 
-      assert to_keys.([]) == [:args, :queue, :state]
-      assert to_keys.(worker: MyWorker) == [:args, :queue, :state, :worker]
+      keys_with_defaults = ~w(args attempt errors max_attempts queue state)a
 
-      assert to_keys.(schedule_in: 1, worker: MyWorker) == [
-               :args,
-               :queue,
-               :scheduled_at,
-               :state,
-               :worker
-             ]
+      default_keys_with = fn extra_keys ->
```

Comment:
Nice solution. Can we pull this (and the existing `to_keys` function) out into private functions?